cosipy/image_deconvolution/RichardsonLucy.py

```diff
@@ -172,15 +193,15 @@ def post_processing(self):
 
         if self.mask is not None:
             self.model = self.model.mask_pixels(self.mask)
-        
-        # update expectation_list
-        self.expectation_list = self.calc_expectation_list(self.model, dict_bkg_norm = self.dict_bkg_norm)
-        logger.debug("The expected count histograms were updated with the new model map.")
```

Comment:
Can you keep these lines? Otherwise, the likelihood is evaluated with the old model.